scripts/complexity.rb

```diff
@@ -0,0 +1,100 @@
+# @author Anton Hößl <e1427045@student.tuwien.ac.at>
+# @brief This script approximates the complexity of a elektra configuration namespace
+# @date 18.04.2018
+# @tags generator
+
+require 'kdb'
+
+# function checks if the type of the key is known and returns the number of bits needed to represent all possible values
+# of that key, returns 1 if it is a unknown type. Doesn't consider range checks etc...
+def evaluate_type(t, tc, k)
```

Comment:
Please use more descriptive variable names. For example, instead of `t` you could use `type` or something similar.